python/tests/test_variable.py

```diff
@@ -163,14 +167,216 @@ def test_create_1D_vector_3_float64():
     assert var.unit == sc.units.m
 
 
-def test_create_datetime64():
-    var = sc.Variable(dims=['x'],
-                      unit=sc.units.ns,
-                      shape=[2],
-                      dtype=sc.dtype.datetime64)
-    assert var.dims == ['x']
-    assert var.dtype == sc.dtype.datetime64
-    assert var.unit == sc.units.ns
+def test_construct_0d_datetime():
+    for unit in ('s', 'us', 'ns'):
+        dtype = f'datetime64[{unit}]'
+        value = np.datetime64(2431, unit)
+        # with value
+        for var in (sc.Variable(dtype=dtype, unit=unit, value=value),
+                    sc.Variable(unit=unit, value=value),
+                    sc.Variable(dtype=dtype,
+                                value=value), sc.Variable(value=value)):
+            assert str(var.dtype) == 'datetime64'
+            assert var.unit == unit
+            assert var.value.dtype == dtype
+            assert var.value == value
+        # default init
+        for var in (sc.Variable(dtype=dtype,
+                                unit=unit), sc.Variable(dtype=dtype)):
+            assert str(var.dtype) == 'datetime64'
+            assert var.unit == unit
+            assert var.value.dtype == dtype
+
+
+def test_construct_0d_datetime_mismatch():
+    for unit1, unit2 in filter(
+            lambda t: t[0] != t[1],
+            itertools.product(('s', 'us', 'ns'), ('s', 'us', 'ns'))):
+        with pytest.raises(ValueError):
+            sc.Variable(unit=unit1, dtype=f'datetime64[{unit2}]')
+        with pytest.raises(RuntimeError):
+            sc.Variable(value=np.datetime64('now', unit1),
+                        dtype=f'datetime64[{unit2}]')
+        with pytest.raises(RuntimeError):
+            sc.Variable(value=np.datetime64('now', unit1), unit=unit2)
+
+
+def test_0d_datetime_setter():
+    for unit in ('s', 'us', 'ns'):
+        initial = np.datetime64(np.random.randint(0, 1000), unit)
+        value = np.datetime64(np.random.randint(0, 1000), unit)
+        var = sc.Variable(value=initial)
+        var.value = value
+        assert var.value == value
```

Comment:
Redundant?

Reply:
The whole test or just one assertion?

Reply:
The assertion.

Reply:
Yeah, I guess so given that we test the getter elsewhere.